### DISCUSSION TITLE
Create PR for new feature

### DISCUSSION THREAD

**Discussion by NuwanSameera** [rabbitmq/rabbitmq-server#5295] [Other]
I created new feature in rabbitmq management plugin to delete all connections by username. It is tested in local, inside docker and the Kubernetes 3 node cluster.

I took branch from last stable v3.10.6 TAG and did my changes and committed to my fork repository.  I wish to create PR to remote repository. Before that I need to create PR from my feature branch to master. But there are lots of conflicts between v3.10.6 and master branch. 

What are steps I need to follow to create PR to remote repository.

**Comment by lukebakken** [comment]
You need to rebase your feature branch **onto** master:

https://git-scm.com/docs/git-rebase

https://womanonrails.com/git-rebase-onto

Then, open your PR against `master`. But, before you do all of that continue reading...

Based on the description of your feature [here](https://github.com/rabbitmq/rabbitmq-server/discussions/5251#discussioncomment-3187409) we probably won't accept it into RabbitMQ as it is very specific to your use-case. @michaelklishin thoughts?

**Reply by NuwanSameera** [reply]
OK. Please let me know RabbitMQ accept this feature. If yes I will create PR.

Thank you for the help and guide to implement my use-case.

**Reply by michaelklishin** [reply]
We would consider a feature that allows connections to be closed by username but by far the most convenient way to us to assess what you are proposing is to see a PR. We cannot guarantee that it would be accepted but the feature doesn't sound like something we would instantly reject.

**Reply by NuwanSameera** [reply]
Correction : Please consider following branch

https://github.com/NuwanSameera/rabbitmq-server/tree/feature-delete-connection-by-username 

And followings are the changes diff

https://github.com/rabbitmq/rabbitmq-server/compare/v3.10.6...NuwanSameera:rabbitmq-server:feature-delete-connection-by-username

**Comment by lukebakken** [comment]
@NuwanSameera I see you opened this PR -

https://github.com/rabbitmq/rabbitmq-server/pull/5318

This is the correct command to rebase your branch _from_ `v3.10.6` _onto_ `master`:

```
git rebase --onto master v3.10.6 feature-delete-connection-by-username
```

You can see that it produces the expected result here:

https://github.com/rabbitmq/rabbitmq-server/compare/master...lukebakken:rabbitmq-server:feature-delete-connection-by-username

**Reply by NuwanSameera** [reply]
I created new PR 
https://github.com/rabbitmq/rabbitmq-server/pull/5319

Please check it is fine.

**Reply by lukebakken** [reply]
Got it!
